File: C++/Selection_Sort.cpp
//Selection Sort 
#include <bits/stdc++.h> 
using namespace std; 

void swap(int *x, int *y) 
{ 
	int temp = *x; 
	*x = *y; 
	*y = temp; 
} 

void selectionSort(int arr[], int n) 
{ 
	int i, j, min; 

	// One by one move boundary of unsorted subarray 
	for (i = 0; i < n-1; i++) 
	{ 
		// Find the minimum element in unsorted array 
		min = i; 
		for (j = i+1; j < n; j++) 
		if (arr[j] < arr[min]) 
			min = j; 

		// Swap the minimum element with the first element 
		swap(&arr[min], &arr[i]); 
	} 
} 


int main() 
{ 
	int arr[] = {10, 22, 15, 45, 39,40}; 
	int n = sizeof(arr)/sizeof(arr[0]); 
	selectionSort(arr, n); 
	cout << "Sorted array: \n"; 
	for (int i=0; i < n; i++) 
		cout << arr[i] << " "; 
	cout << endl; 
	return 0; 
} 

File: python/selection_sort.py
'''
Selection sort algorithm implimented in Python
for more info on selection sort visit:
https://en.wikipedia.org/wiki/Selection_sort
'''

def selection_sort(arr):
	for i in range(len(arr)):
		minimum = i

		for j in range(i+1, len(arr)):
			if arr[j] < arr[minimum]:
				minimum = j;

		if minimum != i:
			arr[i], arr[minimum] = arr[minimum], arr[i]

if __name__ == '__main__':
	arr = [5, 10, 2, 0, 78, -2]
	selection_sort(arr)
	print(arr)

'''
output: [-2, 0, 2, 5, 10, 78]
'''